### DISCUSSION TITLE
Get info from filestream

### DISCUSSION THREAD

**Discussion by endasil** [Zeugma440/atldotnet#176] [Q&A] [answered]
Is there something special i have to think about when reading files from streams? It seems like for any audio file i try i just get 0 values, while if i enter a path to the file the values seem fine.

using var fileStream = System.IO.File.OpenRead("C:\\testwav.wav");
Track audioInfo = new(fileStream, "audio/wav");
audioInfo2.Duration is 0

Track audioInfo = new("C:\\testwav.wav");
Duration is 493

**Comment by Zeugma440** [comment]
You don't seem to do anything wrong here.

I've just used the unit test below with a WAV file and the `"audio/wav"` MIME type successfuly.

https://github.com/Zeugma440/atldotnet/blob/4bdece1af7ece65b31ed3be346845e65ce9ce9f8/ATL.test/IO/HighLevel.cs#L950

Are you sure you're not doing anything else with your stream between creating it and giving it to ATL ? If so, please reset its position to `0` before feeding it to ATL. ATL won't do it automatically to allow for more flexibility (e.g. reading WAV data within a huge stream of multiple concatenated WAV files)

**Reply by Zeugma440** [reply]
NB : Waiting for your reply on that one before publishing the BitDepth changes, just in case there's something else to push ;-)

**Comment by endasil** [comment, marked as answer]
I'm not sure what I did wrong there, i can't reproduce it. Sorry for bothering you about this. If I run into any issue again ill make sure I create git repo with a working minimal sample including whatever i'm using to test it to make sure it can be reproduced in a clean scenario.

**Reply by Zeugma440** [reply]
No problem. Thanks :-)

**Comment by endasil** [comment]
Okay some more testing and a repo setup. When i run this program https://github.com/endasil/soundloaderror to test the accompanied flac file, I get length 0 when loading with filestream. When I load directly with path i get the length. What could the issue be here?

**Reply by Zeugma440** [reply]
Got that one. I didn't register the `audio/flac` MIME-type but only `audio/x-flac`. That's now fixed in today's v4.15

**Reply by endasil** [reply]
Cool, thank you. If a mime is unknown, would it be better to throw an exception or something to make it clear that this is not supported instead of silently failing to read it?

**Reply by Zeugma440** [reply]
You're completely right. The ATL logger now throws a warning when that happens : d9704820b69691bbfcadcc9f469a534f0bc50d79

Available on ~~next release~~ today's v4.16

**Reply by endasil** [reply]
Thank you for the update. I have another music file that i'm having issues with. It parses it as length 0 and 0 channels when it's a several min long stereo track. Do you have an email address I can send you a link to for feedback? As i'm not the owner of the music I do not want to share it public.

**Reply by Zeugma440** [reply]
Sure. You can send me a link to [my Github username] [at] [not-cold] [mail] [dot com] 😉

**Reply by endasil2** [reply]
Thanks, good coding of mail there. Just need to check with the artist and ill send it to you once he gets back to me. :)

**Reply by Zeugma440** [reply]
Don't tell me I'm gonna have to sign an NDA for that 😒

**Reply by endasil** [reply]
I have sent two wav files to you now, if i managed to decode your email correctly. I hope you're able to figure out the issue is. Don't worry, no one asked for a NDA to be signed, i just think it would be rude to send out other peoples music without asking that they are ok with it.

**Reply by Zeugma440** [reply]
Thanks !  Both have the same issue (super short Label fields inside the LIST chunk). My code wasn't handling them properly until now ;)
